src/Sulu/Bundle/SnippetBundle/Content/SingleSnippetSelection.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SnippetBundle\Content;
+
+use Sulu\Bundle\SnippetBundle\Snippet\DefaultSnippetManagerInterface;
+use Sulu\Bundle\SnippetBundle\Snippet\SnippetResolverInterface;
+use Sulu\Bundle\SnippetBundle\Snippet\WrongSnippetTypeException;
+use Sulu\Bundle\WebsiteBundle\ReferenceStore\ReferenceStoreInterface;
+use Sulu\Component\Content\Compat\PropertyInterface;
+use Sulu\Component\Content\Compat\Structure\PageBridge;
+use Sulu\Component\Content\PreResolvableContentTypeInterface;
+use Sulu\Component\Content\SimpleContentType;
+
+class SingleSnippetSelection extends SimpleContentType implements PreResolvableContentTypeInterface
+{
+    /**
+     * @var SnippetResolverInterface
+     */
+    private $snippetResolver;
+
+    /**
+     * @var DefaultSnippetManagerInterface
+     */
+    private $defaultSnippetManager;
+
+    /**
+     * @var ReferenceStoreInterface
+     */
+    private $snippetReferenceStore;
+
+    public function __construct(
+        SnippetResolverInterface $snippetResolver,
+        DefaultSnippetManagerInterface $defaultSnippetManager,
+        ReferenceStoreInterface $snippetReferenceStore
+    ) {
+        $this->snippetResolver = $snippetResolver;
+        $this->defaultSnippetManager = $defaultSnippetManager;
+        $this->snippetReferenceStore = $snippetReferenceStore;
+
+        parent::__construct('SingleSnippetSelection', null);
+    }
+
+    public function getContentData(PropertyInterface $property)
+    {
+        $resolvedSnippet = $this->resolveSnippet($property);
+
+        if (null === $resolvedSnippet) {
+            return $this->defaultValue;
```

Comment:
Would not use `$this->defaultValue` here instead:

```suggestion
            return  null;
```